contracts/jobs/JobsManager.sol

```diff
@@ -227,11 +224,20 @@ contract JobsManager is ManagerProxyTarget, IVerifiable, IJobsManager {
 
         // Job cannot be inactive
         require(jobStatus(_jobId) != JobStatus.Inactive);
-        // Sender must be elected transcoder
-        require(job.transcoderAddress == msg.sender);
         // Segment range must be valid
         require(_segmentRange[1] >= _segmentRange[0]);
 
+        if (job.transcoderAddress != address(0)) {
+            // If transcoder already assigned, check if sender is
+            // the assigned transcoder
+            require(job.transcoderAddress == msg.sender);
+        } else {
+            // If transcoder is not already assigned, check if sender
+            // should be assigned and that job creation block + 1 has been mined and it has been <= 256 blocks since the job creation block
+            require(block.number > job.creationBlock + 1 && block.number <= job.creationBlock + 256 && bondingManager().electActiveTranscoder(job.maxPricePerSegment, job.creationBlock + 1) == msg.sender);
```

Comment:
If we're using the block that the job creation was mined in, then we should only check that block.number > job.creationBlock.